opm/simulators/linalg/gpuistl/OpmGpuILU0.cpp

```diff
@@ -80,6 +84,15 @@ OpmGpuILU0<M, X, Y, l>::OpmGpuILU0(const M& A, bool splitMatrix, bool tuneKernel
         m_gpuReorderedLU = detail::createReorderedMatrix<M, field_type, GpuSparseMatrix<field_type>>(
             m_cpuMatrix, m_reorderedToNatural);
     }
+
+    if (m_storeFactorizationAsFloat){
+        assert(m_splitMatrix && "Mixed precision GpuILU0 is currently only supported when using split_matrix=true");
```

Comment:
I would prefer a `OPM_ERROR_IF` here, an exception is a lot easier to reason about and catch.